src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,257 @@
+
+            var capturedTypeParameters = GetCapturedTypeParameters(declaredSymbol);
+            var unusedTypeParameters = capturedTypeParameters.ToList();
+            foreach (var id in localFunction.DescendantNodes().OfType<IdentifierNameSyntax>())
+            {
+                var sym = semanticModel.GetSymbolInfo(id).Symbol;
+                unusedTypeParameters.Remove(sym?.OriginalDefinition);
+            }
+
+            var typeParameters = capturedTypeParameters
+                .Except(unusedTypeParameters)
+                .Reverse()
+                .Cast<ITypeParameterSymbol>()
+                .ToList();
+
+            var captures = dataFlow.Captured.OfType<ILocalSymbol>();
+            var capturesAsParameters = captures.Select(capture => Parameter(
+                attributeLists: default,
+                modifiers: dataFlow.WrittenInside.Contains(capture) ? TokenList(Token(RefKeyword)) : default,
+                type: capture.Type.GenerateTypeSyntax(),
+                identifier: capture.Name.ToIdentifierToken(),
+                @default: default)).ToList();
+
+            var container = localFunction.GetAncestor<MemberDeclarationSyntax>();
+            var generator = CSharpSyntaxGenerator.Instance;
+
+            var typeParameterSyntaxs = typeParameters.Select(tp => (TypeParameterSyntax)tp.DeclaringSyntaxReferences[0].GetSyntax());
+
+            var methodName = GenerateUniqueMethodName(declaredSymbol).ToIdentifierToken();
+            SyntaxNode method = MethodDeclaration(
+                attributeLists: default,
+                modifiers: GetModifiers(isStatic),
+                returnType: localFunction.ReturnType,
+                explicitInterfaceSpecifier: default,
+                identifier: methodName,
+                typeParameterList: GetTypeParameterList(typeParameterSyntaxs),
+                parameterList: GetParameterList(localFunction, capturesAsParameters),
+                constraintClauses: default,
+                body: localFunction.Body != null ? Block(localFunction.Body.Statements) : null,
+                expressionBody: localFunction.ExpressionBody,
+                semicolonToken: default);
+
+            foreach (var tp in typeParameters)
+            {
+                method = WithTypeConstraint(method, tp, generator);
+            }
+
+            var editor = new SyntaxEditor(root, generator);
+            editor.InsertAfter(container, method);
+            editor.RemoveNode(localFunction, SyntaxRemoveOptions.KeepNoTrivia);
+
+            var enclosingTypeParameters = typeParameters.Except(declaredSymbol.GetTypeParameters()).ToList();
+
+            var supportsNonTrailing = SupportsNonTrailingNamedArguments(root.SyntaxTree.Options);
+
+            // Update callers' arguments and type arguments
+            foreach (var currentInvocation in parentBlock.DescendantNodes().OfType<InvocationExpressionSyntax>())
```

Comment:
Not just invocations.  Someone may also just be referencing the local function directly.  i.e.:

```c#
int Foo(int i) { }
Func<int, int> f = Foo;
```

If you do something like rename Foo then youneed to capture this as well.